vowpalwabbit/fb_parser/src/parse_example_flatbuffer.cc

```diff
@@ -201,26 +201,51 @@ void parser::parse_namespaces(VW::workspace* all, example* ae, const Namespace*
   auto& fs = ae->feature_space[index];
 
   if (hash_found) { fs.start_ns_extent(hash); }
-  for (const auto& feature : *(ns->features()))
-  { parse_features(all, fs, feature, (all->audit || all->hash_inv) ? ns->name() : nullptr); }
-  if (hash_found) { fs.end_ns_extent(); }
-}
 
-void parser::parse_features(VW::workspace* all, features& fs, const Feature* feature, const flatbuffers::String* ns)
-{
-  if (flatbuffers::IsFieldPresent(feature, Feature::VT_NAME))
+  if(ns->feature_names() != nullptr)
```

Comment:
Hashing is expensive.  Wondering if the following is better.
1) if hashes exist, use that.  if strings also exist, then add audit info.  [see point 3]
2) if only strings exist, hash(string), add audit info. [see point 3]
3) audit info addition should only be triggered when audit flag is passed into vw